### DISCUSSION TITLE
How to add Asset Bundles to the Configuration?

### DISCUSSION THREAD

**Discussion by Karottenrambo** [jelte/be.khepri.play.assetdelivery.addressables#57] [Q&A]
Im stuck with this since 2 days :/

When I add the folder, where I copied all the Assetbundle files that the Addressable Build script created, I get an error message: "This folder is missing an AssetBundle file that matches the folder name"

![folder missing assetbundle](https://user-images.githubusercontent.com/17185301/182107930-cf709524-2f75-4397-9c32-b0aa80db16d4.jpg)

I already tried several variations with diffrent build paths, creating custom Asset Pack folder and assetbundle settings, not even the sample project from unity works.

Does anyone have an idea?

**Comment by a-l-e-x-d-s-9** [comment]
Hi,

I'm not sure what's the problem in your case. I don't touch or copy Assetbundle files...
From my experience, it is very important to choose proper versions for everything. The following versions that worked for me:
1. Unity Addressable Package version 1.18.9, any version above never works for me. I have opened a question here regarding supported versions, but never got any response.
2. Google Play Plugins for Unity 1.5 / 1.6 / 1.7 (1.8 is not working for me).
3. Not sure regarding other versions of Unity, these versions worked for me: 2020.3.25f1-2020.3.36f1.

Also make sure that you are doing the sequence:
1. First, Addressable Groups > Build > Default Build Script.
2. Then you need to run Google > Create config for Addressable Groups.
4. Then you build the aab with Google > Build Android app bundle...

Attaching screenshots of my configurations, maybe they will help you: 
![Screenshot from 2022-08-01 14-07-45](https://user-images.githubusercontent.com/6518064/182136176-0770438a-ff58-4a3f-a204-19d894cf33fc.png)
![Screenshot from 2022-08-01 14-09-51](https://user-images.githubusercontent.com/6518064/182136182-1ec531f9-e2e0-416e-9912-bcacd639dd02.png)
![Screenshot from 2022-08-01 14-10-36](https://user-images.githubusercontent.com/6518064/182136184-9f6ef588-817b-4eb2-a0d0-10ffc0899323.png)
![Screenshot from 2022-08-01 14-11-54](https://user-images.githubusercontent.com/6518064/182136188-dd3327ba-eb1d-403a-ae1f-a5e5a8e9e2de.png)

**Reply by EmpEazy** [reply]
> Hi,
> 
> I'm not sure what's the problem in your case. I don't touch or copy Assetbundle files... From my experience, it is very important to choose proper versions for everything. The following versions that worked for me:
> 
> 1. Unity Addressable Package version 1.18.9, any version above never works for me. I have opened a question here regarding supported versions, but never got any response.
> 2. Google Play Plugins for Unity 1.5 / 1.6 / 1.7 (1.8 is not working for me).
> 3. Not sure regarding other versions of Unity, these versions worked for me: 2020.3.25f1-2020.3.36f1.
> 
> Also make sure that you are doing the sequence:
> 
> 1. First, Addressable Groups > Build > Default Build Script.
> 2. Then you need to run Google > Create config for Addressable Groups.
> 3. Then you build the aab with Google > Build Android app bundle...
> 
> Attaching screenshots of my configurations, maybe they will help you: ![Screenshot from 2022-08-01 14-07-45](https://user-images.githubusercontent.com/6518064/182136176-0770438a-ff58-4a3f-a204-19d894cf33fc.png) ![Screenshot from 2022-08-01 14-09-51](https://user-images.githubusercontent.com/6518064/182136182-1ec531f9-e2e0-416e-9912-bcacd639dd02.png) ![Screenshot from 2022-08-01 14-10-36](https://user-images.githubusercontent.com/6518064/182136184-9f6ef588-817b-4eb2-a0d0-10ffc0899323.png) ![Screenshot from 2022-08-01 14-11-54](https://user-images.githubusercontent.com/6518064/182136188-dd3327ba-eb1d-403a-ae1f-a5e5a8e9e2de.pn. After following the steps, do I need to add the asset bundle folder to the “ Google -> Android App Bundle -> Asset delivery Settings” or should I just go straight to the “Google > Build Android app bundle” step. Thanks

**Comment by Karottenrambo** [comment]
Thanks for your help!

I have the identical versions as you have, just on the addressable Package I'm already at 1.19.19.
The only significant change that I had was that I did not enable "Build Remote Catalogue"

But unfortunately this does not change anything with my initial problem.

After I the addressable script I cannot Run Google -> "Create config for Addressables Groups"

I get an Error : Unable to create AssetPack config. Make sure Addressables asset bundles have been build!

![no bundles](https://user-images.githubusercontent.com/17185301/182155671-70e5b822-adb9-401b-84ce-75b061424d96.jpg)

I can only assume that I need to add the Asset bundle Folders in the menu Google -> Android App Bundle -> Asset delivery Settings first, but I'm not able to do so...

**Reply by a-l-e-x-d-s-9** [reply]
Addressable Package 1.19.19 did not work for me, so I assume you downgraded to 1.18.9...
You can try to do: Addressable Groups > Build > Clean, and then build . You can try to to restart the PC.
I would not suggest you to clean your tmp folders, it is very painful when Unity rebuilding everything, but there is tiny chance that it will change something...
I'm not sure what else you can do. Maybe you can ask @jelte for help, he responded to somebody else today, so maybe he is not too far.

**Comment by Karottenrambo** [comment]
The issue really was the addressables version 1.19.19

The downgrade was a bit of a pain because of the dependencies, but right after I installed everything again, I was able to click "Create config for Addressables Groups" without having any folders selected in the Asset delivery settings.

The ABB bundle now built without isses, I hope the rest works out too. :)

Thanks for your help alex! 
@jelte might still want to look into this since its not working with the current addressables package.

**Reply by a-l-e-x-d-s-9** [reply]
Cheers
